base/ltshipout.dtx

```diff
@@ -1789,11 +1789,13 @@
 %    that condition and also check if
 %    \cs{@kernel@after@shipout@lastpage} contains any code. If both
 %    are empty we omit the page generation.
+%    \changes{v1.0p}{2025/04/27}{Increment page counter gh/1717}
 %    \begin{macrocode}
           \bool_lazy_and:nnF
             { \hook_if_empty_p:n {shipout/lastpage} }
             { \tl_if_empty_p:N \@kernel@after@shipout@lastpage }
             {
+              \advance\c@page\@ne
```

Comment:
Might be correct to change locally here but if so it probably warrants a comment